File: cpp/0637二叉树的层平均值.cpp
#include<vector>
#include"mybitree.h"
#include<deque>

vector<double> averageOfLevels(TreeNode *root)
{
    vector<double> res;
    deque<TreeNode *> nodes;

    nodes.push_back(root);

    while (!nodes.empty())
    {
        double s = 0;
        int count = 0;

        int size = nodes.size();

        for (int i = 0; i < size; ++i)
        {
            TreeNode *crt = nodes.front();
            nodes.pop_front();
            if (!crt)
                continue;
            
            ++count;
            s += crt->val;
            nodes.push_back(crt->left);
            nodes.push_back(crt->right);
        }
        if (count)
            res.push_back(s/count);
    }

    return res;
}

File: python/main/0637二叉树的层平均值.py
from binarytree import TreeNode
from collections import deque

def averageOfLevels(root:TreeNode) -> list:
    res=[]

    nodes=deque([root])

    while(nodes):
        size=len(nodes)
        level_vals=[]

        for _ in range(size):
            crt=nodes.popleft()

            if(crt):
                level_vals.append(crt.val)
                nodes.append(crt.left)
                nodes.append(crt.right)
        
        if(level_vals):
            res.append(round(sum(level_vals)/len(level_vals),5))
        

    return res